pkg/ovalutil/dpkg.go

```diff
@@ -102,44 +113,66 @@ func DpkgDefsToVulns(ctx context.Context, root *oval.Root, protoVulns ProtoVulns
 						}
 						consts := root.Variables.ConstantVariables[i]
 						for _, v := range consts.Values {
-							if pkg, ok := pkgcache[v.Body]; !ok {
-								p := &claircore.Package{
-									Name: v.Body,
-									Kind: claircore.BINARY,
-								}
-								pkgcache[v.Body] = p
-								vuln.Package = p
-							} else {
-								vuln.Package = pkg
-							}
+							ns = append(ns, v.Body)
+						}
+					} else {
+						ns = append(ns, name.Body)
+					}
+					for _, n := range ns {
+						vuln := *protoVuln
+						if state != nil {
+							vuln.FixedInVersion = state.EVR.Body
 							if state.Arch != nil {
 								vuln.ArchOperation = mapArchOp(state.Arch.Operation)
 								vuln.Package.Arch = state.Arch.Body
 							}
 						}
-						vulns = append(vulns, &vuln)
-						// early continue
-						continue
-					}
-
-					if pkg, ok := pkgcache[name.Body]; !ok {
-						p := &claircore.Package{
-							Name: name.Body,
-							Kind: claircore.BINARY,
+						if pkg, ok := pkgcache[n]; !ok {
+							p := &claircore.Package{
+								Name: name.Body,
+								Kind: claircore.BINARY,
+							}
+							pkgcache[n] = p
+							vuln.Package = p
+						} else {
+							vuln.Package = pkg
 						}
-						pkgcache[name.Body] = p
-						vuln.Package = p
-					} else {
-						vuln.Package = pkg
-					}
-					if state.Arch != nil {
```

Comment:
do we no longer want to add the arch values to the vuln?

Reply:
The code still does, just in a way that allow for the `state` object to be nil.